lib/elixir/lib/tuple.ex

```diff
@@ -128,6 +128,22 @@ defmodule Tuple do
     :erlang.delete_element(index + 1, tuple)
   end
 
+  @doc """
+  Computes a product of tuple elements.
+
+  ## Examples
+
+      iex> tuple = {255, 255}
+      iex> Tuple.product(tuple)
+      65025
+  """
+  @doc since: "1.12.0"
+  @spec product(tuple) :: integer()
```

Comment:
We also allow:

```elixir
iex> Tuple.product({2.0, 3}) #=> 6.0
```

so I believe the spec should be:

```suggestion
  @spec product(tuple) :: number()
```